monix-execution/jvm/src/main/scala/monix/execution/tracing/CorrelationId.scala

```diff
@@ -0,0 +1,16 @@
+package monix.execution.tracing
+
+import monix.execution.misc.{TracingContext, TracingContextCompanion}
+
+/**
+  * A tracing context that allows to propagate a correlation id
+  * through an execution.
+  * @param id the correlation id.
+  */
+case class CorrelationId(id: String) extends TracingContext {
```

Comment:
Please make all `case class` definitions that you do `final`, because they are not and there's no reason for why they shouldn't be.